ci/mac_ci_setup.sh

```diff
@@ -12,9 +15,9 @@ function is_installed {
 
 function install {
     echo "Installing $1"
-    if ! brew install "$1"
-    then
-        echo "Failed to install $1"
+    brew install "$1"
+    if ! brew link --overwrite "$1"; then
```

Comment:
Is this necessary if we are unlinking below?

Reply:
this overwrite was added in order to overwrite existing `bazelisk` symlinks that homebrew complained about (see the output here before the forceful link: https://dev.azure.com/cncf/envoy/_build/results?buildId=33967&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=7d499b5d-1d92-5096-7919-3c7f8065da78&l=619), this change could potentially actually supercede the unlink of `bazel` so this could be the only change needed (will try it out)

Reply:
So, both are required, see this build: https://dev.azure.com/cncf/envoy/_build/results?buildId=33982&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=7d499b5d-1d92-5096-7919-3c7f8065da78&l=619

We could do a `brew install --force` on all installs to maybe not have to unlink bazel, though that may have unintended consequences?

Reply:
it should work, i'll leave it with the `--force` unless theres any objections

Reply:
@sunjayBhatia sorry I didn't catch that `bazelisk` was already installed. Per your comment above, why can't we just drop all the bazel related installs in this file?

Reply:
@junr03 We totally could, but I believe `bazel install bazelisk` downloads something from somwhere different than `bazel install bazelbuild/tap/bazelisk`. If we're cool with staying with what is in the CI image by default that works. However, I don't think if we do that the symlink for `bazel` will actually be the `bazelisk` executable, so we would need to do some manual overriding of symlinks in that case.

Reply:
We could also ask the maintainers of the image to install `bazelisk` (and `bazel`) via the expected upstream tap so this isn't an issue for us as well, seems like at least `bazelisk` is not installed in the image via `bazelbuild/tap/bazelisk`

Reply:
> However, I don't think if we do that the symlink for bazel will actually be the bazelisk executable.

I see, and we would end up using the version of bazel installed in the image instead of the one we would like via bazelisk and .bazelversion.

My overall opinion for this PR is to take bazelisk install out of the normal flow and do what we need to do for bazel/bazelisk separately from the other dep installs. I just don't think we should change how we install all the deps. Now whether this means making sure that bazel is correctly symlinked to bazelisk, or using brew, up to you.